account.py

```
@@ -0,0 +1,85 @@
+class Account:
+    """
+    class for a bank account that has an account number, password and balance, and has methods that:
+    return the password, prints the balance, withdraws and deposits from balance, change the password, get teh account's
+    data and a method dictionary that contains the methods used by the main program.
+    """
+
+    def __init__(self, account_number, password, balance=0):
+        self.account_number = account_number
+        self.__password = password
+        self.__balance = balance
+
+    def __eq__(self, other):
+        return self.account_number == other.account_number
+
+    def __hash__(self):
+        return hash(self.account_number)
+
+    def get_password(self):
+        return self.__password
+
+    def print_balance(self):
+        print(f'\nYour balance is: {self.__balance}')
+
+    def withdraw(self):
+        """
+        asks for withdrawal amount, checks that it's not negative, is a number and not bigger then the account's balance,
+        then subtracts it from the balance.
+        """
+        try:
+            withdrawal_amount = float(input('\nEnter Withdrawal amount: '))
+        except ValueError:
+            raise ValueError('Input must be a number!')
+        if withdrawal_amount > self.__balance:
+            print('Not enough balance!')
+            raise ValueError
+        if withdrawal_amount < 0:
+            print('Withdrawal amount must be positive!')
+            raise ValueError
+        self.__balance -= withdrawal_amount
+
+    def deposit(self):
+        """
+        asks for deposit amount, checks that it's not negative and is a number, then adds it to the account's balance.
+        """
+        try:
+            deposit_amount = float(input('\nEnter deposit amount: '))
+        except ValueError:
+            print('Input must be a number!')

```

can you just raise ValueError('Input must be a number!') instead of printing?

---

I now see that it crashes the program. do you catch the value error?